http/src/v0/refs/path.rs

```diff
@@ -133,7 +133,7 @@ impl IpfsPath {
                 // FIXME: this would require the iterator to be peekable in addition.
                 return Ok(WalkSuccess::Link(key, cid));
             }
-            Ipld::Map(mut m) if m.contains_key(&key) => {
+            Ipld::Map(mut m) => {
```

Comment:
Aah of course perhaps I refactored this back and forth and forgot. I don't think error is ever checked. Btw, here the error sucks, moving stuff into the error is ... too much, no-one probably ever needs the partial documents and those might be too large to print either way.